Tests/Test_Renode_MotorSimulator.c

```diff
@@ -0,0 +1,136 @@
+#include "os.h"
+#include "CANbus.h"
+#include <stdio.h>
+#include "Tasks.h"
+
+// Global variables
+static float velocity; 
+static float current; 
+static float total_accel;
+static CANDATA_t oldCD;
+static int ctr = 0;
+static float motor_force;
+static int forceLUTIndex;
+
+// FORCE LUT 
+// FORCELUT Size
+#define FORCELUT_SIZE 21 
+// Force Lookup Table: Indexed by Current%toIndex, returns a force applied by current % (in Newtons)
+const int FORCELUT[FORCELUT_SIZE] = {
+    0,
+    235,
+    471,
+    706,
+    941,
+    1176,
+    1412,
+    1647,
+    1882,
+    2118,
+    2353,
+    2588,
+    2824,
+    3059,
+    3294,
+    3529,
+    3765,
+    4000,
+    4235,
+    4471,
+    4706
+};
+
+// Macros for calculation the velocity of the car
+#define MS_TIME_DELAY_MS 100
+#define DECELERATION 2.0
+#define CAR_MASS_KG 270
+
+static OS_TCB Task1TCB;
+static CPU_STK Task1Stk[DEFAULT_STACK_SIZE];
+
+// Helper function to convert from current to desired FORCELUT index
+inline int currentPercentToIndex(float currentPercent){
+        return ((int)(currentPercent*FORCELUT_SIZE));
+    }
+
+// Helper function to convert time delay from ms to s
+inline float millisToS(int timeInMS) {
+    return (((float)MS_TIME_DELAY_MS) / 1000);
```

Comment:
There might be a better way to do this with fixed point

Reply:
no. 😏